ingestion/src/metadata/sampler/sampler_interface.py

```diff
@@ -0,0 +1,246 @@
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Interface for sampler
+"""
+import traceback
+from abc import ABC, abstractmethod
+from typing import Dict, List, Optional, Set, Union
+
+from metadata.generated.schema.entity.data.database import Database
+from metadata.generated.schema.entity.data.databaseSchema import DatabaseSchema
+from metadata.generated.schema.entity.data.table import (
+    ColumnProfilerConfig,
+    Table,
+    TableData,
+)
+from metadata.generated.schema.entity.services.connections.connectionBasicType import (
+    DataStorageConfig,
+)
+from metadata.generated.schema.entity.services.connections.database.datalakeConnection import (
+    DatalakeConnection,
+)
+from metadata.generated.schema.entity.services.databaseService import DatabaseConnection
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.profiler.api.models import TableConfig
+from metadata.profiler.processor.sample_data_handler import upload_sample_data
+from metadata.sampler.config import (
+    get_exclude_columns,
+    get_include_columns,
+    get_profile_sample_config,
+    get_sample_data_count_config,
+    get_sample_query,
+)
+from metadata.sampler.models import SampleConfig
+from metadata.sampler.partition import get_partition_details
+from metadata.utils.constants import SAMPLE_DATA_DEFAULT_COUNT
+from metadata.utils.execution_time_tracker import calculate_execution_time
+from metadata.utils.logger import sampler_logger
+from metadata.utils.sqa_like_column import SQALikeColumn
+from metadata.utils.ssl_manager import get_ssl_connection
+
+logger = sampler_logger()
+
+
+class SamplerInterface(ABC):
+    """Sampler interface"""
+
+    # pylint: disable=too-many-instance-attributes, too-many-arguments
+    def __init__(
+        self,
+        service_connection_config: Union[DatabaseConnection, DatalakeConnection],
+        ometa_client: OpenMetadata,
+        entity: Table,
+        include_columns: Optional[List[ColumnProfilerConfig]] = None,
+        exclude_columns: Optional[List[str]] = None,
+        sample_config: SampleConfig = SampleConfig(),
+        partition_details: Optional[Dict] = None,
+        sample_query: Optional[str] = None,
+        storage_config: DataStorageConfig = None,
+        sample_data_count: Optional[int] = SAMPLE_DATA_DEFAULT_COUNT,
+        **__,
+    ):
+        self.ometa_client = ometa_client
+        self._sample = None
+        self._columns: Optional[List[SQALikeColumn]] = None
+        self.sample_config = sample_config
+
+        if not self.sample_config.profile_sample:
+            self.sample_config.profile_sample = 100
```

Comment:
I think we should enforce sample type = percent if we'll do that.

Reply:
will check

Reply:
the default for the sampling type is already percentage. Do you have some sane default in the case it's ROW?